src/main/scala/is/hail/utils/IntervalTree.scala

```diff
@@ -21,10 +21,31 @@ case class Interval(start: Any, end: Any, includeStart: Boolean, includeEnd: Boo
       false
   }
 
+  def includes(pord: ExtendedOrdering, other: Interval): Boolean =
+  other.definitelyEmpty(pord) || ({
+    val cstart = pord.compare(this.start, other.start)
+    cstart < 0 || ((!other.includeStart || this.includeStart) && cstart == 0)
```

Comment:
may as well move `cstart == 0` left of &&, here and elsewhere, since it's more likely to short circuit